dotnet/src/Experimental/Process.UnitTests/ProcessStepEdgeBuilderTests.cs

```diff
@@ -0,0 +1,150 @@
+    }
+
+    /// <summary>
+    /// Verify that the <see cref="ProcessStepEdgeBuilder.SendEventTo(ProcessFunctionTargetBuilder)"/> method throws if the output target is already set.
+    /// </summary>
+    [Fact]
+    public void SendEventToShouldThrowIfOutputTargetAlreadySet()
+    {
+        // Arrange
+        var source = new ProcessStepBuilder<TestStep>(TestStep.Name);
+        var eventType = "Event1";
+        var builder = new ProcessStepEdgeBuilder(source, eventType);
+        var outputTarget1 = new ProcessFunctionTargetBuilder(source);
+        var outputTarget2 = new ProcessFunctionTargetBuilder(source);
+
+        // Act
+        builder.SendEventTo(outputTarget1);
+
+        // Assert
+        Assert.Throws<InvalidOperationException>(() => builder.SendEventTo(outputTarget2));
+    }
+
+    /// <summary>
+    /// Verify that the <see cref="ProcessStepEdgeBuilder.StopProcess"/> method sets the output target to the end step.
+    /// </summary>
+    [Fact]
+    public void StopProcessShouldSetOutputTargetToEndStep()
+    {
+        // Arrange
+        var source = new ProcessStepBuilder<TestStep>(TestStep.Name);
+        var eventType = "Event1";
+        var builder = new ProcessStepEdgeBuilder(source, eventType);
+
+        // Act
+        builder.StopProcess();
+
+        // Assert
+        Assert.Equal(EndStep.Instance, builder.OutputTarget?.Step);
+    }
+
+    /// <summary>
+    /// Verify that the <see cref="ProcessStepEdgeBuilder.StopProcess"/> method throws if the output target is already set.
+    /// </summary>
+    [Fact]
+    public void StopProcessShouldThrowIfOutputTargetAlreadySet()
+    {
+        // Arrange
+        var source = new ProcessStepBuilder<TestStep>(TestStep.Name);
+        var eventType = "Event1";
+        var builder = new ProcessStepEdgeBuilder(source, eventType);
+        var outputTarget = new ProcessFunctionTargetBuilder(source);
+
+        // Act
+        builder.SendEventTo(outputTarget);
+
+        // Assert
+        Assert.Throws<InvalidOperationException>(() => builder.StopProcess());
+    }
+
+    /// <summary>
+    /// Verify that the <see cref="ProcessStepEdgeBuilder.Build"/> method returns a <see cref="KernelProcessEdge"/>.
+    /// </summary>
+    [Fact]
+    public void BuildShouldReturnKernelProcessEdge()
+    {
+        // Arrange
+        var source = new ProcessStepBuilder<TestStep>(TestStep.Name);
+        var eventType = "Event1";
+        var builder = new ProcessStepEdgeBuilder(source, eventType);
+        var outputTarget = new ProcessFunctionTargetBuilder(source);
+        builder.SendEventTo(outputTarget);
+
+        // Act
+        var edge = builder.Build();
+
+        // Assert
+        Assert.NotNull(edge);
+        Assert.Equal(source.Id, edge.SourceStepId);
+    }
+
+    /// <summary>
+    /// A class that represents a step for testing.
+    /// </summary>
+    public class TestStep : KernelProcessStep<TestState>
```

Comment:
Making this `private` will eliminate the demand for comments (imho - this test mock doesn't require much commenting)